ambry-api/src/main/java/com/github/ambry/config/RouterConfig.java

```diff
@@ -777,5 +780,15 @@ public RouterConfig(VerifiableProperties verifiableProperties) {
         ROUTER_NOT_FOUND_CACHE_MAX_TTL_IN_MS);
     routerUpdateOpMetadataRelianceTimestampInMs = verifiableProperties.getLong(
         ROUTER_UPDATE_OP_METADATA_RELIANCE_TIMESTAMP_IN_MS, DEFAULT_ROUTER_UPDATE_OP_METADATA_RELIANCE_TIMESTAMP_IN_MS);
+
+    compressionConfig = new CompressionConfig(verifiableProperties);
+  }
+
+  /**
+   * Get the compression-related configs.
+   * @return The compression-related configs.
+   */
+  public CompressionConfig getCompressionConfig() {
+    return compressionConfig;
```

Comment:
we should consider https://projectlombok.org/features/GetterSetter

Reply:
Yes, we should use lombok.  It saves us from writing getters and setters.

Reply:
awesome ! these getters and setters are mostly noise.

Reply:
Do note that lombok can have its own issues. If an IDE doesn't support lombok, then the code will show up as error in the IDE. We are open source code, so not sure if tying ourselves to certain IDEs is a good idea. As lombok releases new versions, if they are backward incompatible, the IDEs will need sometime to update their support.
Also putting breakpoints and creating javadocs will be hard. Sometimes code reading could also become harder, because you won't be able to find the method you are looking for

Not saying that we should not consider lombok. But lets do make sure we understand the pros and cons, and weigh it considering that we are open source. Maybe we can strictly define the cases where we use lombok (only pure POJOs?), and ensure that our style checker can validate that.